### PR TITLE
[REEF-1741] Make a single (global) nuget.

### PR DESCRIPTION
**Pull request by Emeka-NMC**
This addressed the issue by:
  * creating a single REEF Nuget package that contains the contents-of (and not just references-to) the other 11 REEF Nuget packages.

JIRA:
  [REEF-1741] (https://issues.apache.org/jira/browse/REEF-1741)

Pull request:
  This closes REEF-1741

### CODE REVIEW

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    <title>Org.Apache.REEF</title>
+    <authors>The Apache REEF project</authors>
+    <owners>The Apache REEF project</owners>
+    <licenseUrl>http://www.apache.org/licenses/LICENSE-2.0</licenseUrl>
+    <projectUrl>http://reef.apache.org/</projectUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Provide single entry point that contains all necessary REEF/WAKE/TANG projects.</description>
+    <copyright>The Apache Software Foundation</copyright>
+    <tags>Single entry point to reference all Reef/Wake/Tang projects </tags>
+    <dependencies>
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec]: Why would all of these be mentioned as dependencies again? I thought the idea was to package the DLLs from these into a single NuGet?

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<package >
+  <metadata>
+    <id>Org.Apache.REEF</id>
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec]: I don't think this NuGet should be called just `Org.Apache.REEF`. Maybe `Org.Apache.REEF.FatNuGet`? In java land, we call these things "fat" jars. But maybe there is a better name?

Comment (jwang98052) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec]: Agree. We should not use just REEF. FatNuGet is OK to me.

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec (hunk 1 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec]: Looks like removing those REEF dependencies still works as we have packaged all the dlls.

**Comment by jwang98052** [comment]
You can add Org.Apache.REEF.IMRU as a reference in csproj file for completeness.

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.csproj [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    </ProjectReference>
+    <ProjectReference Include="$(SolutionDir)\Org.Apache.REEF.Utilities\Org.Apache.REEF.Utilities.csproj">
+      <Project>{79e7f89a-1dfb-45e1-8d43-d71a954aeb98}</Project>
+      <Name>Org.Apache.REEF.Utilities</Name>
+    </ProjectReference>
+    <ProjectReference Include="$(SolutionDir)\Org.Apache.REEF.Wake\Org.Apache.REEF.Wake.csproj">
+      <Project>{cdfb3464-4041-42b1-9271-83af24cd5008}</Project>
+      <Name>Org.Apache.REEF.Wake</Name>
+    </ProjectReference>
+  </ItemGroup>
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.csproj]: Please add Org.Apache.REEF.IMRU as a reference for completeness.

**Comment by jwang98052** [comment]
I have completed a pass. Please update and push the change to your branch again.

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec (hunk 2 above)
Comment (Emeka-NMC) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec]: OK; renamed.

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec (hunk 1 above)
Comment (Emeka-NMC) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec]: Mistake; removed.

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.csproj (hunk 3 above)
Comment (Emeka-NMC) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.csproj]: OK, added missing Org.Apache.REEF.IMRU & Org.Apache.REEF.IO.

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Client\Org.Apache.REEF.Client.dll" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Client\Org.Apache.REEF.Client.pdb" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Common\Org.Apache.REEF.Common.dll" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Common\Org.Apache.REEF.Common.pdb" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Driver\Org.Apache.REEF.Driver.dll" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Driver\Org.Apache.REEF.Driver.pdb" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Evaluator\NugetExeFix.txt" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Evaluator\Org.Apache.REEF.Evaluator.exe" target="lib\net45" />
+    <!-- TODO[REEF-1620]: Remove the below -->
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Evaluator\Org.Apache.REEF.Evaluator.exe" target="tools" />
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec]: remove this line and TODO comment line as it was for backward compatibility of a previous change.

File: lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      <dependency id="Org.Apache.REEF.Utilities" version="$version$" />
+      <dependency id="Org.Apache.REEF.Wake" version="$version$" />
+      <dependency id="protobuf-net" version="2.0.0.668" />
+      <dependency id="Rx-Core" version="2.2.5" />
+    </dependencies>
+  </metadata>
+  <files>
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.All\Org.Apache.REEF.All.dll" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.All\Org.Apache.REEF.All.pdb" target="lib\net45" />
+    <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.Client\Org.Apache.REEF.Client.dll" target="lib\net45" />
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF/Org.Apache.REEF.nuspec]: Try to reference all the projects needed that would ensure those dlls are available in the bin folder of this project after build. Then change the reference here to pick up those dlls from the current project bin folder ..\bin\$Platform$\$Configuration$\Org.Apache.REEF\...  . That would make this project self contained.

**Pull request closed without merging**